ionoscloud/data_source_dnsaas_record.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	dnsaas "github.com/ionos-cloud/sdk-go-dnsaas"
+	"log"
+	"strings"
+)
+
+func dataSourceDNSaaSRecord() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceRecordRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "The ID of your DNS Record.",
+				Optional:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of your DNS Record.",
+				Optional:    true,
+			},
+			"partial_match": {
+				Type:        schema.TypeBool,
+				Description: "Whether partial matching is allowed or not when using name argument.",
+				Default:     false,
+				Optional:    true,
+			},
+			"zone_id": {
+				Type:        schema.TypeString,
+				Description: "The UUID of an existing DNS Zone",
+				Required:    true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"content": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ttl": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"fqdn": {
+				Type:        schema.TypeString,
+				Description: "Fully qualified domain name",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func dataSourceRecordRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).DNSaaSClient
+	partialMatch := d.Get("partial_match").(bool)
+	zoneId := d.Get("zone_id").(string)
+	idValue, idOk := d.GetOk("id")
+	nameValue, nameOk := d.GetOk("name")
+	recordId := idValue.(string)
+	recordName := nameValue.(string)
+
+	if idOk && nameOk {
+		return diag.FromErr(fmt.Errorf("ID and name cannot be both specified at the same time"))
+	}
+	if !idOk && !nameOk {
+		return diag.FromErr(fmt.Errorf("please provide either the DNS Record ID or name"))
+	}
+	if partialMatch && !nameOk {
+		return diag.FromErr(fmt.Errorf("partial_match can only be used together with the name attribute"))
+	}
+
+	var record dnsaas.RecordResponse
+	var err error
+
+	if idOk {
+		record, _, err = client.GetRecordById(ctx, zoneId, recordId)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("an error occured while fetching the DNS Record with ID: %s, DNS Zone ID: %s, error: %w", recordId, zoneId, err))
+		}
+	} else {
+		var results []dnsaas.RecordResponse
+		log.Printf("[INFO] Populating data source for DNS Record using name: %s and partial_match: %t", recordName, partialMatch)
+		if partialMatch {
+			// By default, when providing the name as a filter, for the GET requests, partial match
+			// is true.
+			records, _, err := client.ListRecords(ctx, zoneId, recordName)
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("an error occured while fetching DNS Records: %w", err))
+			}
+			if records.Items != nil {
```

Comment:
maybe just break if records.Items == nil and the error will be handled by `if results == nil || len(results) == 0 {`

Reply:
@cristiGuranIonos done in 01e2b65f9fe5d96e6c21367e7fe3b2e7b6343ba2, I completely removed it.